### PR TITLE
Add service for converting diff to grade based on grading system

### PR DESCRIPTION
**Pull request by NejcS**
This PR adds service `GradingSystemsService` which will be used to convert between grades and difficulties based on provided grading system.

I implemented method `diffToGrade` which returns the appropriate grade name and modifier for the provided difficulty. It can also convert to legacy grades but only if the provided difficulty is "accurate" (400, 425, 450 ...).

I think it should work for all grading systems because it loops over the array of grades instead of doing arithmetics. But I didn't test other than french yet because I want to confirm that this is the way to go first before I put in more time.

I noticed that this service converts diffs to grades differently than the Grade class but I think that this is accurate now.

**Example using `Grade.ts` class:**
Screenshot 2021-12-22 at 20 24 46

**Example using `GradingSystemsService`:**
Screenshot 2021-12-22 at 20 24 23

**Difference between old and new implementation**
Difficulty 716.6 would get displayed as 5a (not soft or hard) but now it's a hard 5a.
Difficulty 484 would get displayed as 4b (not soft or hard) but now it's a soft 4b.

I made this for easier development:
image

### CODE REVIEW

File: src/app/common/grade/grade.component.ts [hunk 1]
```diff
@@ -12,14 +13,19 @@ export class GradeComponent implements OnInit {
...
   modifier: number = 0;
 
   gradeLetters = ['a', 'a+', 'b', 'b+', 'c', 'c+'];
 
-  constructor() {}
+  constructor(private GradingSystemsService: GradingSystemsService) {}
 
   ngOnInit(): void {
-    this.grade = new Grade(this.difficulty);
+    // this.grade = new Grade(this.difficulty);
```
Comment (NejcS) [inline src/app/common/grade/grade.component.ts:25]: Uncomment this and comment out `this.grade = name;` to see the old implementation if you want to quickly compare the differences.

And then also uncomment the import and property definition above.

File: src/app/common/grade/grade.component.html [hunk 2]
```diff
@@ -1,8 +1,10 @@
-<span class="grade-name">{{ legacy ? grade.legacyName : grade.name }}<ng-container *ngIf="showModifier"> </ng-container>
```
Comment (salamca) [inline src/app/common/grade/grade.component.html:1]: ngif on an empty container, is this a mistake?

<ng-container *ngIf="showModifier"> </ng-container>

Comment (NejcS) [reply src/app/common/grade/grade.component.html:1]: It was there before and I was guessing that it is there because of the space (" ") to seaprate the grade name from the modifier.

But I didn't check that.

**Review by salamca — APPROVED** [review]
Looks good. 
Did some tests also on uiaa and yds. 
Also checked some corner cases on soft/hard/full rendering.

Maybe would suggest displaying some kind of invalid grade msg/string if difficulty is out of range?
(less than 100 or more than 2100)
